pages/api/checkout/session.ts

```diff
@@ -16,21 +16,46 @@ export default async function sessionHandler(
     switch (req.method) {
         case "POST": {
             const stripeRequest = req.body as StripeCheckoutRequest;
-            const session = await stripe.checkout.sessions.create({
-                payment_method_types: ["card"],
-                line_items: [
-                    {
-                        price: stripeRequest.priceId,
-                        quantity: stripeRequest.quantity,
-                    },
-                ],
-                mode: "payment",
-                success_url: `${req.headers.origin}/result?session_id={CHECKOUT_SESSION_ID}`,
-                cancel_url: `${req.headers.origin}/checkout`,
-                allow_promotion_codes: true,
-            });
+            // if a coupon ID is passed in then apply an automatic discount
+            if (stripeRequest.couponId) {
```

Comment:
This is small but if we wanted to make the code cleaner, we could first use an object to store the data being created. Then, we use this if condition to conditionally set `discounts` in the object. After the else statement, we could have `const session = await stripe.checkout.sessions.create(checkoutObject)`. Just a suggestion to reduce the amount of duplicate/similar code! Don't need to change if you disagree.

Reply:
Great suggestion! Done.